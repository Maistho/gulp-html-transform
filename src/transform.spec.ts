import { testTransform } from './test/helpers'
import { readFileSync } from 'fs'
import { join } from 'path'

describe('gulp-html-transform', () => {
  it(`doesn't change anything if there is no transforms`, async () => {
    const filename = 'test.html'
    const content = await testTransform(filename, [])
    const fileContent = readFileSync(
      join(__dirname, 'test', filename),
    ).toString()
    expect(content).toBe(fileContent)
  })
  it(`can accept a transformer`, async () => {
    const transformer = async ($: any) => {
      $('img').attr('src', 'https://i.imgur.com/HObogkM.gif')
    }
    const content = await testTransform('image.html', [transformer])
    expect(content).toMatchSnapshot()
  })
})
