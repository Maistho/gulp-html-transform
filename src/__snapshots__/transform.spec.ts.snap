// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gulp-html-transform can accept a transformer 1`] = `
"<html><head></head>
  <body>
    <img src=\\"https://i.imgur.com/HObogkM.gif\\">
  

</body></html>"
`;
